Bumped version to 8.8.6

tpr-martyn-hills opened the pull request:
Added the ability to override the section card title headings level so it can be set from a h1 to a h6 so that a page can follow the correct structure for semantic design.

Updated example and documentation for the new feature.

tpr-martyn-hills committed:
Bumped version to 8.8.6

Added the ability to override the section card title headings level so it can be set from a h1 to a h6 so that a page can follow the correct structure for semantic design.

Updated example and documentation for the new feature.

sussexrick commented on GovUk.Frontend.ExampleApp/Views/SectionCards/Index.cshtml:
```diff
@@ -19,10 +19,14 @@
             <p class="govuk-body">The section cards component displays a list of cards for navigation, usually linking to child pages.</p>
+
+            <p class="govuk-body">
+                The heading level of all the titles can be set from 1 to 6 using the <strong>card-titles-heading-level</strong> parameter.
```
I don't think h1 will ever be appropriate in this scenario, and probably not the lower levels either. Rather than use the existing 'GOV.UK Heading level' data type I would suggest creating a custom data type for this scenario similar to 'GOV.UK Heading level (Accordion)'.

sussexrick commented on ThePensionsRegulator.Frontend/ITprHtmlGenerator.cs:
```diff
@@ -15,7 +14,7 @@ public interface ITprHtmlGenerator
         TagBuilder GenerateTprContextBar(TprContextBar tprContextBar);
         TagBuilder GenerateTprRelatedLinks(TprRelatedLinks tprRelatedLinks);
-        TagBuilder GenerateTprSectionCards(TprSectionCards tprSectionCards);
+        TagBuilder GenerateTprSectionCards(int sectionCardsTitleHeadingLevel, TprSectionCards tprSectionCards);
```
Add a property to `TprSectionCards` rather than a new parameter on the method.

sussexrick commented on ThePensionsRegulator.Frontend.Umbraco/Views/Shared/TPR/TPRSectionCards.cshtml:
```diff
@@ -17,17 +17,25 @@
-    
+
+    string headingLevelProperty = Model.Settings?.Value<string>(TprPropertyAliases.SectionCardsTitleHeadingLevel) ?? string.Empty;
+    var headingLevel = headingLevelProperty.Replace("heading", "", System.StringComparison.OrdinalIgnoreCase);
```
Very minor point as it's clearly working, but the replacement should be for "heading " (with a space) rather than "heading".

sussexrick commented on ThePensionsRegulator.Frontend.Umbraco/Views/Shared/TPR/TPRSectionCards.cshtml:
```diff
@@ -17,17 +17,25 @@
+        numericHeadingLevel = 2;
+    }
+
+    <tpr-section-cards class="@cssClasses" new-tab-text="@Umbraco.GetDictionaryValueOrDefault(DictionaryConstants.OpensInNewTab, DictionaryConstants.OpensInNewTab)" card-titles-heading-level="@numericHeadingLevel">
```
The appearance is different for each heading level (h3 is appearing smaller than h2 on your example page). I think you need to update the `tpr-section-card__title` CSS class to specify the same font size as is used for h2 by default.

tpr-martyn-hills replied:
Added the ability to set the titles heading class using the same pattern as the Search Results feature.

tpr-martyn-hills committed:
Added support for card title heading class in section cards

Introduce `card-titles-heading-class` parameter to `tpr-section-cards`, enabling customization of CSS classes for card title headings. Supported classes include `govuk-heading-xl`, `govuk-heading-l`, `govuk-heading-m`, and `govuk-heading-s`, with a default of `govuk-heading-m`.

Updated documentation and assets:
- Updated `tpr-section-cards.md` to document the new parameter.
- Updated `tpr-section-cards-settings-options.png` to reflect changes.
- Added `TPRHeadingLevelSectionCards.config` to define valid heading level options.

These changes improve flexibility and consistency in the `tpr-section-cards` component.

tpr-martyn-hills committed:
Merge branch 'develop' into bugfix/243450-section-cards

tpr-martyn-hills committed:
Updated documentation

sussexrick commented on ThePensionsRegulator.Frontend/HtmlGeneration/TprSectionCards.cs:
```diff
@@ -8,5 +8,7 @@ public record TprSectionCards
         public AttributeDictionary? Attributes { get; set; }
         public List<TprSectionCard> Cards { get; set; } = new();
         public string? NewTabText { get; set; }
+        public int sectionCardsTitleHeadingLevel = 2;
```
We shouldn't ever use public fields, so these should be PascalCase properties.

tpr-martyn-hills replied:
Apologies as a typo and all updated

tpr-martyn-hills committed:
Add navigation aria-label to section cards

Introduced a `navigation-aria-label` parameter to the `<tpr-section-cards>` component to enhance accessibility by allowing developers to set an `aria-label` attribute on the `nav` element. If left empty, no `aria-label` is rendered.

- Refactored `TprSectionCards.cs` to use PascalCase and added `NavigationAriaLabel`.
- Updated `tpr-section-cards-settings-options.png` to reflect the new setting.

Refactored code for consistency and improved readability.

AB#240624
AB#243450

sussexrick committed:
Merge branch 'develop' into bugfix/243450-section-cards
